waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -242,15 +197,15 @@ proc reconnectPeers*(pm: PeerManager,
   
   debug "Reconnecting peers", proto=proto
   
-  for storedInfo in pm.peers(protocolMatcher):
+  for storedInfo in pm.peerStore.peers(protocolMatcher):
     # Check if peer is reachable.
-    if pm.peerStore.connectionBook[storedInfo.peerId] == CannotConnect:
+    if pm.peerStore[ConnectionBook][storedInfo.peerId] == CannotConnect: #TODO what if it doesnt exist?
```

Comment:
I'm also curious about this TODO. :D

Reply:
default(T), which is debatable
```nim
if storedInfo.peerId in pm.peerStore[ConnectionBook] and pm.peerStore[ConnectionBook][storedInfo.peerId] == CannotConnect:

# or a pattern I start to like, even though it's ugly, {} returning an option
if pm.peerStore[ConnectionBook]{storedInfo.peerId} == some(CannotConnect)
```

Reply:
I love snippets that can be run `nim r file.nim`. I'm surprised with the result, a bit dangerous isn't it?

```nim
import libp2p/peerstore
import libp2p/peerid
let myPeerStore = PeerStore.new(capacity = 5)

type ConnectionBook = ref object of PeerBook[string]

var p1, p2: PeerId
discard p1.init("peer1")
discard p2.init("peer2")

myPeerStore[ConnectionBook][p1] = "Connected"

echo myPeerStore[ConnectionBook][p1]
echo myPeerStore[ConnectionBook][p2]

#Output:
#Connected
#Connected
```